ivy_tests/test_ivy/test_frontends/test_mindspore/test_ops/test_function/test_mindspore_nn_func.py

```diff
@@ -77,3 +77,51 @@
 #         on_device=on_device,
 #         x=x[0],
 #     )
+
+# interpolate
+# @handle_frontend_test(
+#     fn_tree="mindspore.ops.function.nn_func.interpolate",
+#     dtype_and_x=helpers.dtype_and_values(
+#         available_dtypes=helpers.get_dtypes("valid"),
+#         num_arrays=1,
+#         shared_dtype=True,
+#         min_value=2,
+#         max_value=5,
+#         min_dim_size=4,
+#         shape=(
+#             st.integers(min_value=2, max_value=10),
+#             4,
+#             st.integers(min_value=12, max_value=64),
+#             st.integers(min_value=12, max_value=64),
+#         ),
+#     ),
+#     size=st.lists(st.integers(min_value=1, max_value=10), min_size=3, max_size=3) | None,
```

Comment:
The size argument can also accept `int` and `tuple[int]` types, which should also be tested here. `scale_factor`, similarly, supports `float` and `tuple[float]`.

Also, how do you enforce that only one of them can be `None`?

Reply:
It appears that both `size` and `scale_factor` are allowed to be None. The `| None` part in both lines makes the respective argument optional, meaning it can take either the specified values or `None`. There is no enforcement that only one of them can be `None`.

To clarify, here's what the code is doing for each argument:

1. `size=st.lists(st.integers(min_value=1, max_value=10), min_size=3, max_size=3) | None,`

   - The `size` argument is allowed to be a list of three integers with values between 1 and 10 (inclusive) and must have a minimum size of 3 and a maximum size of 3.
   - Alternatively, it can be set to `None`, making it an optional argument.

2. `scale_factor=st.lists(st.floats(min_value=0.1, max_value=2.0), min_size=3, max_size=3) | None,`

   - The `scale_factor` argument is allowed to be a list of three floats with values between 0.1 and 2.0 (inclusive) and must have a minimum size of 3 and a maximum size of 3.
   - Alternatively, it can be set to `None`, making it an optional argument.

There is no enforcement that only one of them can be `None`. Both `size` and `scale_factor` can be provided or set to `None` independently. They are independent optional arguments.

Reply:
Sorry for the misunderstanding. The restriction I was talking about is from the mindspore docs I linked above:
>  One and only one of size and scale_factor can be set to None.